Source/DafnyLanguageServer/Workspace/FoundFiles.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Dafny.LanguageServer.Workspace.Notifications;
+using Microsoft.Extensions.Logging;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.Dafny.LanguageServer.Workspace;
+
+record FoundFiles(
```

Comment:
Why do we have these records?

Reply:
Is your question why is this a record instead of a class? Because it saves boilerplate in creating fields.

Or is your question, why is there a compilation event called FoundFiles? That's because the start of the compilation involves evaluating and validating the paths and globs specified in the CLI and in the project file, and this event exposes the results of that, which provides useful information for the IDE.

Reply:
My question was merely the second one. Thanks for answering both!